notebooks/official/workbench/spark/spark_sample_notebook.ipynb

```diff
@@ -0,0 +1,1200 @@
+    "- Ingesting data from BigQuery into a Spark DataFrame.\n",
+    "- Preprocessing ingested data.\n",
+    "- Querying the most frequently used programming language in monoglot repos.\n",
+    "- Querying the average size (MB) of code in each language stored in monoglot repos.\n",
+    "- Querying the languages files most frequently found together in polyglot repos.\n",
+    "- Writing the query results back into BigQuery.\n",
+    "- Deleting the resources created for this notebook tutorial.\n",
+    "- Disabling the APIs used in the tutorial."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "tvgnzT1CKxrO"
+   },
+   "source": [
+    "### Costs \n",
+    "\n",
+    "This tutorial uses billable components of Google Cloud:\n",
+    "\n",
+    "* [Vertex AI](https://cloud.google.com/vertex-ai/pricing)\n",
+    "* [Cloud Storage](https://cloud.google.com/storage/pricing)\n",
+    "* [Dataproc](https://cloud.google.com/dataproc/pricing)\n",
+    "\n",
+    "You can use the [Pricing Calculator](https://cloud.google.com/products/calculator/) to generate a cost estimate based on your projected usage."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "BF1j6f9HApxa"
+   },
+   "source": [
+    "## Before you begin\n",
+    "\n",
+    "### Set up your Google Cloud project:\n",
+    "\n",
+    "1. [Select or create a Google Cloud project](https://console.cloud.google.com/cloud-resource-manager). When you create an account, you receive a $300 credit towards to your compute and storage costs.\n",
+    "\n",
+    "1. [Make sure that billing is enabled for your project](https://cloud.google.com/billing/docs/how-to/modify-project).\n",
+    "\n",
+    "1. [Enable Notebooks API, Vertex AI API, and Dataproc API](https://console.cloud.google.com/flows/enableapi?apiid=notebooks.googleapis.com,aiplatform.googleapis.com,dataproc&_ga=2.209429842.1903825585.1657549521-326108178.1655322249)\n",
+    "\n",
+    "**Note**: Jupyter runs lines prefixed with `!` as shell commands, and inserts the value of Python variables prefixed with `$` into the commands."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "lWEdiXsJg0XY"
+   },
+   "source": [
+    "### Create a Dataproc cluster\n",
+    "\n",
+    "The Spark job executed in this notebook tutorial is compute intensive. Since the job can take a significant amount time to complete in a standard notebook environment, this notebook tutorial runs on a Dataproc cluster that is created with the Dataproc Component Gateway and Jupyter component installed on the cluster.\n",
+    "\n",
+    "**Existing Dataproc with Jupyter cluster?**: If you have a running Dataproc cluster you can use for this tutorial (it must have the [Component Gateway and Jupyter component installed on the cluster](https://cloud.google.com/dataproc/docs/concepts/components/jupyter#gcloud-command)), you can skip this and go to `Switch your kernel`."
```

Comment:
Suggest: If you have a running Dataproc cluster that has the [Component Gateway and Jupyter component installed on the cluster (https://cloud.google.com/dataproc/docs/concepts/components/jupyter#gcloud-command)), you can use it this tutorial. If you plan to use it, skip this step, and go to  `Switch your kernel`.